grid/pdf-export-server/Client/Services/PdfExportService.cs

```diff
@@ -0,0 +1,56 @@
+﻿using Microsoft.AspNetCore.Components;
+using Microsoft.JSInterop;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Json;
+using System.Threading.Tasks;
+using Telerik.DataSource;
+
+namespace ServerPdfExport.Client.Services
+{
+    public class PdfExportService
+    {
+        [Inject]
+        private HttpClient Http { get; set; }
+
+        [Inject]
+        private IJSRuntime JS { get; set; }
+
+        public PdfExportService(HttpClient client, IJSRuntime _js)
+        {
+            Http = client;
+            JS = _js;
+        }
+
+        public async Task GetPdf(DataSourceRequest gridRequest, bool allPages, bool useExcelGeneration)
+        {
+            //if we want all the pages, remove the paging from the grid request
+            //you can use any other means of transporting and using this information
+            if (allPages)
+            {
+                gridRequest.PageSize = 0;
```

Comment:
The [request.Skip](https://github.com/telerik/blazor/blob/4e5255719346b4bd34798d68ad66f252efea6949/Telerik.Blazor/Components/Grid/TelerikGrid.razor.cs#L2375) parameter can also be used, so that it covers the virtualization scenario.